Description: Use correct ELF header for armhf binaries.
 Based on Adam Conrad's 016-armhf-elf-header.patch. As Ubuntu only
 builds for armhf, we don't need the conditionality in that patch. A
 real fix is more complex and should be attempted for Go 1.6.
Forwarded: not-needed
Bug: https://github.com/golang/go/issues/7094
Author: Michael Hudson-Doyle <michael.hudson@canonical.com>
Last-Update: 2015-07-24

--- a/src/cmd/link/internal/ld/elf.go
+++ b/src/cmd/link/internal/ld/elf.go
@@ -781,7 +781,7 @@
 	case '5':
 		// we use EABI on both linux/arm and freebsd/arm.
 		if HEADTYPE == obj.Hlinux || HEADTYPE == obj.Hfreebsd {
-			ehdr.flags = 0x5000002 // has entry point, Version5 EABI
+			ehdr.flags = 0x5000402 // has entry point, Version5 EABI, hard-float ABI
 		}
 		fallthrough
 
